### PR TITLE
Edit application event and no person creation

### PR DESCRIPTION
**Pull request by auramo**

### CODE REVIEW

File: src/cljs/ataru/virkailija/application/view.cljs [hunk 1]
```diff
@@ -172,15 +174,20 @@
         [:div.application-handling__review-header "Tila"]]
        (mapv (partial review-state-row @review-state) application-review-states)))))
 
+(defn event-caption [event]
+  (match (:event-type event)
```
Comment (elahti) [inline src/cljs/ataru/virkailija/application/view.cljs]: Maybe `(case (:event-type event) ...)` would be enough here? `clojure.core.match/match` version is 2.9 times as slow as `clojure.core/case`.

File: src/cljs/ataru/hakija/application_handlers.cljs [hunk 2]
```diff
@@ -66,24 +74,24 @@
...
                 :post-data     (create-application-to-submit (:application db) (:form db) (get-in db [:form :selected-language]))
                 :handler       :application/handle-submit-response
-                :error-handler :application/handle-submit-error}
-     :dispatch  [:application/clear-error]}))
+                :error-handler :application/handle-submit-error}}))
+
+(reg-event-fx
+  :application/edit
+  (fn [{:keys [db]} _]
+    (println "EDIT FORM SUBMIT POST STUB")
```
Comment (elahti) [inline src/cljs/ataru/hakija/application_handlers.cljs]: This `println` accidentally left unremoved?

File: src/cljs/ataru/hakija/application_handlers.cljs [hunk 3]
```diff
@@ -66,24 +74,24 @@
...
-    (dissoc db :error)))
+             (assoc :error {:message "Tapahtui virhe " :detail response}))}))
 
 (reg-event-fx
-  :application/submit-form
+  :application/submit
   (fn [{:keys [db]} _]
-    {:db       (assoc-in db [:application :submit-status] :submitting)
+    (send-application db :submit)
+    {:db       (-> db (assoc-in [:application :submit-status] :submitting) (dissoc :error))
```
Comment (elahti) [inline src/cljs/ataru/hakija/application_handlers.cljs]: Is the `dissoc` necessary? Is there a case where we'll end up in `:application/submit` with `:error` something else than `nil`?

File: src/cljs/ataru/hakija/application_handlers.cljs (hunk 2 above)
Comment (auramo) [inline src/cljs/ataru/hakija/application_handlers.cljs]: No, I absolutely want it to remain there!

**Pull request merged**